doc/building.md

```diff
@@ -1947,9 +2064,52 @@ In the JDK build, we always use Unix paths internally, and only just before
 calling a tool that does not understand Unix paths do we convert them to
 Windows paths.
 
-This conversion is done by the `fixpath` tool, which is a small wrapper that
-modifies unix-style paths to Windows-style paths in command lines. Fixpath is
-compiled automatically by `configure`.
+This conversion is done by the `fixpath.sh` tool, which is a small wrapper that
+modifies Unix-style paths to Windows-style paths. The fixpath
+tool is called with the first argument as a sub-command describing the action
+it should take. Available actions are `import`, `exec`, `print` and `verify`.
+
+* `import` is called at configure time to convert a path given by the user and
+  that might be in Windows format to Unix path, which is used internally.
+
+* `exec` is called at build time. This will take the command line provided,
+  complete with arguments, converting the paths in the command line, and then
+  execute the command.
+
+* `print` is called at build time, in the rare cases where a path might be
+  needed in Windows format, but not as an argument to a command to execute.
+
+* `verify` is called at configure time to check that a path is correctly
+  specified and reachable by Windows.
+
+The fixpath tool uses a somewhat complex heuristic to infer which part of the
+command line arguments refer to paths, and converts those. In some
+circumstances, these heuristics can fail.
+
+If you are having strange build issues related to path conversion, you might
+need to debug how fixpath treats your paths. There are many ways to do this.
```

Comment:
many -> "several", or maybe "multiple"? Or perhaps "Here are some ways to do this."